File: schemas/email-campaign-marketing/drizzle/segments.ts
// segments: Dynamic contact groups defined by filter criteria.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index, jsonb } from "drizzle-orm/pg-core";
import { users } from "./users";

export const segments = pgTable(
  "segments",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    description: text("description"),
    filterCriteria: jsonb("filter_criteria").notNull(),
    createdBy: uuid("created_by").references(() => users.id, { onDelete: "set null" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_segments_created_by").on(table.createdBy),
  ],
);

File: schemas/email-campaign-marketing/spacetimedb/segments.rs
// segments: Dynamic contact groups defined by filter criteria.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = segments, public)]
pub struct Segment {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,
    pub description: Option<String>,

    pub filter_criteria: String, // JSON

    #[index(btree)]
    pub created_by: Option<String>, // UUID, FK → users.id (set null)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
